src/Hungary/Mews.Fiscalizations.Hungary/Models/Invoice/Company.cs

```diff
@@ -0,0 +1,17 @@
+﻿using FuncSharp;
+
+namespace Mews.Fiscalizations.Hungary.Models
+{
+    public sealed class Company : Coproduct2<LocalCompany, ForeignCompany>
```

Comment:
Do we actually need the type for this? I wanted to just check the country in the serialization. We'd need to decide if address or the identifier country is the deciding factor. But technically we're deciding that here in the static create anyway - you can't create a LocalCompany if the taxId isn't hungarian, not the address.

Reply:
I would keep it just because it makes the code more readable, setting values based on the data provided would seem like some magic.

Reply:
I'd keep the two classes. I like that you can't create LocalCompany without TaxPayerId.